text_recognizer/data/iam.py

```diff
@@ -37,91 +40,108 @@ class IAM:
-        return _get_ids_from_lwitlrt_split_file(EXTRACTED_DATASET_DIRNAME / "task/testset.txt")
-
-    @cachedproperty
-    def validation_ids(self):
-        """
-        Return a list of IAM lines Large Writer Independent Text Line Recognition Task validation (set 1 and set 2) ids.
-        """
-        val_ids = _get_ids_from_lwitlrt_split_file(EXTRACTED_DATASET_DIRNAME / "task/validationset1.txt")
-        val_ids.extend(_get_ids_from_lwitlrt_split_file(EXTRACTED_DATASET_DIRNAME / "task/validationset2.txt"))
-        return val_ids
-
-    @cachedproperty
-    def train_ids(self):
-        """Return a list of train ids - all ids which aren't test or validation ids."""
-        return list(set(self.all_ids) - (set(self.test_ids) | set(self.validation_ids)))
+    def ids_by_split(self):
+        return {"train": self.train_ids, "val": self.validation_ids, "test": self.test_ids}
 
     @cachedproperty
     def split_by_id(self):
+        """A dictionary mapping form IDs to their split according to IAM Lines LWITLRT."""
         split_by_id = {id_: "train" for id_ in self.train_ids}
         split_by_id.update({id_: "val" for id_ in self.validation_ids})
         split_by_id.update({id_: "test" for id_ in self.test_ids})
         return split_by_id
 
     @cachedproperty
-    def ids_by_split(self):
-        return {"train": self.train_ids, "val": self.validation_ids, "test": self.test_ids}
+    def train_ids(self):
+        """A list of form IDs which are in the IAM Lines LWITLRT training set."""
+        return list(set(self.all_ids) - (set(self.test_ids) | set(self.validation_ids)))
 
     @cachedproperty
-    def word_strings_by_id(self):
-        """Return a dict from name of IAM form to a list of line texts in it."""
-        return {filename.stem: _get_word_strings_from_xml_file(filename) for filename in self.xml_filenames}
+    def test_ids(self):
+        """A list of form IDs from the IAM Lines LWITLRT test set."""
+        return _get_ids_from_lwitlrt_split_file(EXTRACTED_DATASET_DIRNAME / "task/testset.txt")
+
+    @property
+    def xml_filenames(self) -> List[Path]:
+        """The filenames of all .xml files, which contain label information."""
+        return list((EXTRACTED_DATASET_DIRNAME / "xml").glob("*.xml"))
 
     @cachedproperty
-    def word_regions_by_id(self):
-        """Return a dict from name of IAM form to a list of (x1, x2, y1, y2) coordinates of all lines in it."""
-        return {filename.stem: _get_word_regions_from_xml_file(filename) for filename in self.xml_filenames}
+    def validation_ids(self):
+        """A list of form IDs from IAM Lines LWITLRT validation sets 1 and 2."""
+        val_ids = _get_ids_from_lwitlrt_split_file(EXTRACTED_DATASET_DIRNAME / "task/validationset1.txt")
+        val_ids.extend(_get_ids_from_lwitlrt_split_file(EXTRACTED_DATASET_DIRNAME / "task/validationset2.txt"))
+        return val_ids
+
+    @property
```

Comment:
why shouldn't these by `@cachedproperty` ?

Reply:
i think that change was yours originally -- i torpedoed the git blame though when i merged your branches into this one, so it's not immediately obvious

Reply:
my bad, but don't you think they should be `cachedproperty`? or just save it to a variable?

Reply:
hmm, im not entirely sure what the reasoning is for these `cachedproperty`s from `boltons`, but i do know that swapping them in for properties breaks everything, so im leaving them be